src/evict/evict_lru.c

```diff
@@ -1838,14 +1837,16 @@ __evict_walk_file(WT_SESSION_IMPL *session,
 		if (__wt_ref_is_root(ref) || evict == start || give_up ||
 		    ref->page->read_gen == WT_READGEN_OLDEST ||
 		    ref->page->memory_footprint >= btree->splitmempage) {
-			btree->evict_ref = ref;
-			WT_RET(__evict_clear_walk(session, restarts == 0));
-		} else if (ref->page->read_gen == WT_READGEN_OLDEST) {
+			if (restarts == 0)
+				WT_STAT_CONN_INCR(
+				    session, cache_eviction_walks_abandoned);
+			WT_RET(__wt_page_release(cache->walk_session,
```

Comment:
Switching the order of clearing ``evict_ref`` and releasing the page is contrary to the comment in ``__evict_clear_walk`` that reads:

```
  * Clear evict_ref first, in case releasing it forces eviction (we
  * assert we never try to evict the current eviction walk point).
```

Oh, I see now. In the larger change ``evict_ref`` is never set while the tree is actually being walked. I don't think the new comment is quite adequate, but I think the change is OK